kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -56,7 +58,20 @@ private[kyuubi] class EngineRef private(conf: KyuubiConf, user: String, sessionI
   // Share level of the engine
   private val shareLevel: ShareLevel = ShareLevel.withName(conf.get(ENGINE_SHARE_LEVEL))
 
+  // This a is a server-side option
+  val poolThreshold: Int = KyuubiConf().loadFileDefaults().get(ENGINE_POOL_SIZE_THRESHOLD)
```

Comment:
this is not what I mean...

Reply:
:sweat_smile:

Reply:
the right way is to get it from `sessionConf` of KyuubiSessionImpl before merging the client configs.. then the system defaults and user defaults are respected(we can set different threshold for different users)

Reply:
Got.